lhotse/dataset/sampling/weighted_simple.py

```diff
@@ -0,0 +1,243 @@
+import warnings
+from typing import Any, Dict, List, Optional
+
+from lhotse import CutSet, Seconds
+from lhotse.dataset.sampling.base import CutSampler, TimeConstraint
+from lhotse.dataset.sampling.data_source import WeightedDataSource
+
+
+class WeightedSimpleCutSampler(CutSampler):
```

Comment:
I wonder whether this class can inherit `SimpleCutSampler` and only override the necessary parts (e.g. using `WeightedDataSource` in `__init__`); otherwise most of this code looks very similar.

Reply:
Good idea! I just changed the inheritance, please have a look